### DISCUSSION TITLE
How is the partial trace performed for 1d fermions?

### DISCUSSION THREAD

**Discussion by Niklas-Mueller1988** [QuSpin/QuSpin#509] [Q&A]
Computing the reduced DM of a fermionic system is a bit non-trivial: https://arxiv.org/pdf/1211.7217.pdf

How is this actually done in quspin? Is there a Jordan-Wigner transformation "under the hood" and this works because we are in 1d (*), or does quspin explicitly implement an ordering as in Eq.23 of above reference?

(*) This may (or may not) work if one considers the subsystem where the JW string begins (the other subsystem).

**Comment by weinbe58** [comment]
Yes, in version 0.3.6 this issue was resolved. It is done "under the hood" as you say. 

The best way to think about this is the following: The partial trace requires reordering the fock-states such that they respect a new site order where all the subsystem A sites are pushed to the left and the subsystem B sites are pushed to the right. In the case of fermions we also calculate the change in sign for each new fock basis state given the new order of sites relative to the old ordering.

**Reply by Niklas-Mueller1988** [reply]
Yes, that makes sense. I am assuming by system A you mean "the subsystem" and by system B "what is traced over"?

**Reply by weinbe58** [reply]
Yes
